docs/content/topics/project-developers/events.md

```diff
@@ -9,4 +9,229 @@ aliases:
   - /topics/project-developers/events.md
 ---
 
-TODO
\ No newline at end of file
+# Brigade Events
+
+Events are the lingua franca in Brigade: [Gateways] emit events and [Project]
+developers write logic to handle them.
+
+In this document, we'll look at:
+
+  * The [structure of an event]
+  * How to [configure event subscriptions] on a project
+  * How to [handle events] in a Brigade script.
+
+[Gateways]: /topics/operators/gateways
+[Project]: /topics/project-developers/projects
+[structure of an event]: #event-structure
+[configure event subscriptions]: #event-subscriptions
+[handle events]: #handling-events
+
+## Event Structure
+
+A Brigade event is defined primarily by its source and type values. These two
+must be provided every time an event is emitted into Brigade. However, there
+are plenty of other fields that event creators can harness to add additional
+context and information for utilization in handlers.
+
+The full list of fields on an Event is:
+
+  * [Source](#source)
+  * [Type](#type)
+  * [Payload](#payload)
+  * [ProjectID](#project-id)
+  * [Qualifiers](#qualifiers)
+  * [Labels](#labels)
+  * [Short Title](#short-title)
+  * [Long Title](#long-title)
+  * [Source State](#source-state)
+  * [Summary](#summary)
+
+Each field is reviewed in depth below.
+
+### Source
+
+An event's Source can be thought of as its domain. In fact, the rule of thumb
+to avoid clashes is to use a URI you control. This means leading with one's own
+domain or the URL for something else one owns, like the URL for a GitHub repo.
+As an example, events emitted from Brigade's [GitHub Gateway] use a source
+value of `brigade.sh/github`.
```

Comment:
I might be a useful clarification that this rule of thumb is applicable when developing one's own gateway. That context seems to be missing, and without it, this could be confusing.